app/services/owner_classifier.rb

```diff
@@ -3,8 +3,24 @@ def initialize(id_and_type)
     @id_and_type = id_and_type
   end
 
+  def self.classify_owner(id_and_type)
+    return nil if id_and_type.blank?
+
+    validate_input_format!(id_and_type)
+
+    owner_type = id_and_type.end_with?('Users') ? User : Usergroup
+    owner_type.find(id_and_type.to_i)
```

Comment:
Shouldn't this be `find_by_id` like it was? Or `find_by_id!` to raise an exception?

Reply:
Ok, I looked it up myself. https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find states it will call `.to_i` for you so it can even be:
```suggestion
    owner_type.find(id_and_type)
```

But your code is also valid. I don't mind it being explicit here.

Reply:
AFAIK, both `find_by_id!` and `find` raise the same exception when the record is not found (in this case). If you believe that `find_by_id!` enhances readability in this context, I'm open to using it instead.